lib/sparkql/function_resolver.rb

```diff
@@ -118,6 +122,13 @@ def call()
   protected 
   
   # Supported function calls
+
+  def regex(regular_expression)
+    {
+      :type => :character,
+      :value => regular_expression
```

Comment:
From what I've read, we might run into some quirks as between ruby versions, they have changed the underlying lib that handles the regex. Aside from that, it seems simple enough to get back to the string as needed, so I don't feel strongly either.

    1.9.3-p547 :008 > a = Regexp.new "/[A-Z]+/"
     => /\/[A-Z]+\// 
    1.9.3-p547 :009 > a.source
     => "/[A-Z]+/" 

Since we're validating at this point anyways, maybe that's the way to call it.